### PR TITLE
TINY-8155: Add DOM model and move non-UI table logic to model and core

### PR DESCRIPTION
**Pull request by ltrouton**
Related Ticket: TINY-8155

Description of Changes:
* Add model concept to the editor
* Move main table action and commands to the dom model (modules/tinymce/src/models/dom/main/ts/table/api/Commands.ts)
Move fake cell selection handling to core selection (modules/tinymce/src/core/main/ts/api/dom/Selection.ts)
* At the moment, the fake selected cells can be retrieved using editor.selection.getSelectedCells which is what the table plugin and dom model are now using
* Move table resize handling to core selection (modules/tinymce/src/core/main/ts/api/dom/Selection.ts)
* Move tab navigation to core keyboard overrides (modules/tinymce/src/core/main/ts/keyboard/TabKey.ts)
* Move TableFormats to core (modules/tinymce/src/core/main/ts/fmt/FormatRegistry.ts)
* Move `mceInsertTable` command to core (modules/tinymce/src/core/main/ts/api/EditorCommands.ts)
* Remove the need for resize wire (wire) to be passed into Snooker
* Move table tests to where they are most appropriate

Pre-checks:
* [X] Changelog entry added
* [X] Tests have been added (if applicable)
* [X] Branch prefixed with `feature/` for new features (if applicable)

Review:
* [X] Milestone set
* [ ] Review comments resolved

GitHub issues (if applicable):

### CODE REVIEW

**Review by TheSpyder — APPROVED** [review]

File: modules/tinymce/src/plugins/table/main/ts/ui/Buttons.ts [hunk 1]
```diff
@@ -24,160 +30,169 @@ const addButtons = (editor: Editor, selections: Selections, selectionTargets: Se
 
   const cmd = (command: string) => () => editor.execCommand(command);
 
-  editor.ui.registry.addButton('tableprops', {
+  // TODO TINY-8172: unwind this before merging the feature branch
+  const addButtonIfRegistered = (name: string, spec: AddButtonSpec) => {
+    if (editor.editorCommands.hasCustomCommand(spec.command)) {
```
Comment (TheSpyder) [inline modules/tinymce/src/plugins/table/main/ts/ui/Buttons.ts:35]: Due to changes in #7572 please change all of these to `queryCommandSupported`
```suggestion
    if (editor.editorCommands.queryCommandSupported(spec.command)) {
```

**Review by TheSpyder — CHANGES_REQUESTED** [review]

File: modules/tinymce/src/plugins/table/main/ts/ui/Buttons.ts (hunk 1 above)
Comment (ltrouton) [reply modules/tinymce/src/plugins/table/main/ts/ui/Buttons.ts:35]: I have done this as part of #7559

File: modules/tinymce/src/core/main/ts/Rtc.ts [hunk 2]
```diff
@@ -66,6 +67,7 @@ interface RtcRuntimeApi {
     getContent: (args: Partial<GetSelectionContentArgs>) => Content;
   };
   raw: {
+    getSharedModelApi: () => Model;
```
[the remark this answers is not in the document]
Comment (spocke) [reply modules/tinymce/src/core/main/ts/Rtc.ts:70]: It seems that you already did? https://github.com/tinymce/rockfest/pull/498 or is it something else we need?

[the remark this answers is not in the document]
Comment (TheSpyder) [reply modules/tinymce/src/core/main/ts/Rtc.ts:70]: Oh 😅 I forgot I snuck that in there. I should’ve checked!

**Review by TheSpyder — APPROVED** [review]

**Comment by ltrouton** [comment]
Closing this in favour of #7583

**Pull request closed without merging**